legend-engine-xt-nonrelationalStore-mongodb-pure/src/main/resources/core_nonrelational_mongodb/pureToDatabaseCommand/pureToDatabaseCommand.pure

```diff
@@ -66,18 +66,30 @@ function meta::external::store::mongodb::functions::pureToDatabaseCommand::proce
   $functionExpression.parametersValues->at($index)->cast(@StoreMappingRoutedValueSpecification).value->cast(@SimpleFunctionExpression).parametersValues;
 }
 
+function meta::external::store::mongodb::functions::pureToDatabaseCommand::buildFilterPath(simpleFunctionExpression: SimpleFunctionExpression[1], path: String[1]): String[1]
+{
+  if ($simpleFunctionExpression.func != map_T_$0_1$__Function_1__V_$0_1$_,
```

Comment:
I'm not super happy with checking that the func property is not **map_T_$0_1$__Function_1__V_$0_1$_**, but I couldn't find something else to use if the FE.
If the func is **map_T_$0_1$__Function_1__V_$0_1$_**, it means that it has nested levels so we need to go further down, otherwise the func will contain the name of the property we want to use (firstName, lastName etc).